src/twisted/internet/defer.py

```diff
@@ -618,10 +618,12 @@ def addCallback(self, callback: Any, *args: Any, **kwargs: Any) -> "Deferred[Any
 
         See L{addCallbacks}.
         """
-        # Implementation Note: Any annotations for brevity; the overloads above
-        # handle specifying the actual signature, and there's nothing worth
-        # type-checking in this implementation.
-        return self.addCallbacks(callback, callbackArgs=args, callbackKeywords=kwargs)
+        self.callbacks.append(((callback, args, kwargs), (_failthru, (), {})))
```

Comment:
Maybe add a comment that we don't reuse `addCallbacks` to prevent creating another frame.

Also, add a comment that the code from here should be kept in sync with `addErrback` and `addBoth`